docs/csharp/programming-guide/concepts/async/walkthrough-accessing-the-web-by-using-async-and-await.md

```diff
@@ -1,379 +1,369 @@
+    - On the **Text** tab, specify a monospaced font, such as `Consolas` for the **FontFamily**.
+1. Position the text box and the button so that both appear in the **MainWindow** window.
      For more information about the WPF XAML Designer, see [Creating a UI by using XAML Designer](/visualstudio/xaml-tools/creating-a-ui-by-using-xaml-designer-in-visual-studio).
 
-## Add a reference
-
-1. In **Solution Explorer**, highlight your project's name.
-
-2. On the menu bar, choose **Project** > **Add Reference**.
-
-     The **Reference Manager** dialog box appears.
-
-3. At the top of the dialog box, verify that your project is targeting the .NET Framework 4.5 or higher.
-
-4. In the **Assemblies** category, choose **Framework** if it isn't already chosen.
-
-5. In the list of names, select the **System.Net.Http** check box.
+The resulting XAML should resemble the following:
+
+```xaml
+<Window x:Class="AsyncFirstExample.MainWindow"
+        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+        Title="Async Example" Height="400" Width="600" MinHeight="300" MinWidth="500">
+    <Grid>
+        <Button x:Name="_startButton" Content="Start" HorizontalAlignment="Center"
+                Margin="0,10,0,0" VerticalAlignment="Top" Width="75" Height="24" />
+        <TextBox x:Name="_resultsTextBox" TextWrapping="Wrap"
+                 FontFamily="Consolas" VerticalScrollBarVisibility="Visible"
+                 Margin="0,45,0,0" />
+    </Grid>
+</Window>
+```
 
-6. Choose the **OK** button to close the dialog box.
+> [!NOTE]
+> Your `x:Class` name may be different, but should match your corresponding *MainWindow.xaml.cs* file class name.
 
 ## Add necessary using directives
 
-1. In **Solution Explorer**, open the shortcut menu for MainWindow.xaml.cs, and then choose **View Code**.
-
-2. Add the following `using` directives at the top of the code file if they're not already present.
+1. In the **Solution Explorer**, open the shortcut menu for *MainWindow.xaml.cs*, and then choose **View Code**.
+1. Replace all of the `using` directives at the top of the code file with the following:
 
     ```csharp
-    using System.Net.Http;
-    using System.Net;
+    using System.Collections.Generic;
+    using System.Diagnostics;
     using System.IO;
+    using System.Linq;
+    using System.Net;
+    using System.Threading.Tasks;
+    using System.Windows;
     ```
 
 ## Create a synchronous app
 
-1. In the design window, MainWindow.xaml, double-click the **Start** button to create the `startButton_Click` event handler in MainWindow.xaml.cs.
-
-2. In MainWindow.xaml.cs, copy the following code into the body of `startButton_Click`:
+1. In the design window, *MainWindow.xaml*, double-click the **Start** button to create the click event handler in *MainWindow.xaml.cs*.
+1. Select the <kbd>F2</kbd> key to rename the handler to `OnStartButtonClick`.
```

Comment:
Interesting, I think I could have done better by referring to the start button click event handler as just that - rather than the function name. It lends itself to more of the conceptual aspect of documenting the source code, opposed to directly referencing a member. It also future-proofs the doc, so that the code itself can evolve.

But I am against the default naming that the tooling generates. Just because the tooling names something, doesn't mean we have to document it that way. I prefer following best practices in the naming conventions of modern C#. Event handlers should be named `OnObjectEventName`. Method names shouldn't include `_` and should use Pascal casing.

Reply:
Well the point I was bringing out is that our events pattern article for C# is wrong when it says to use `OnObjectEventName`, perhaps @BillWagner can comment on this. [WPF](https://github.com/dotnet/wpf/blob/master/src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/UIElement.cs#L3564), Winforms, and the guidance from the Framework 2.0 design book (which we still have published on the website) says that `OnObjectEventName` is used to raise the events and is implemented in class defining the event as a protected method. It's not supposed to be the event handler name.

Reply:
OK. I was able to look at the new Framework Design Guidelines 3rd Edition book. It does confirm that `OnEventName` should be used for raising events and allowing an inherited class to override that method. The examples it used for adding a handler to event however didn't use `On` or `_` in them. They simply were just named `[EventName]Handler` or `[ObjectNameEventName]`.